src/preamble.js

```diff
@@ -2230,8 +2230,8 @@ function integrateWasmJS(Module) {
   }
 
   function getBinaryPromise() {
-    // if we don't have the binary yet, and have the Fetch api, use that
-    if (!Module['wasmBinary'] && typeof fetch === 'function') {
+    // if we don't have the binary yet, and have the Fetch api, use that, but only when in browser environment
+    if (!Module['wasmBinary'] && (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) && typeof fetch === 'function') {
```

Comment:
overall this looks reasonable to me, thanks. please just add to the comment something like `in some other environments, like Electron, fetch may exist but behave differently than on the Web, so avoid it`.

Reply:
Rephrased, is it better now?